src/lexer.coffee

```diff
@@ -121,7 +121,11 @@ exports.Lexer = class Lexer
         'PROPERTY'
       else
         'IDENTIFIER'
-
+    
+    if tag is 'IDENTIFIER' and @seenFor and id is 'from'
```

Comment:
Well `in` and `of` are reserved keywords, whereas `from` is a context-sensitive identifier. So the lexer can't handle them all in the same way.

```
$ coffee -te 'from'
[IDENTIFIER from] [TERMINATOR \n]

$ coffee -te 'in'
[RELATION in] [TERMINATOR \n]
```

I've merged the `if tag is 'IDENTIFIER'` conditionals into one, maybe that helps it look cleaner.